object_manager.go

```diff
@@ -544,17 +547,64 @@ func (objMgr *ObjectManager) GetARecordByRef(ref string) (*RecordA, error) {
 	err := objMgr.connector.GetObject(recordA, ref, &recordA)
 	return recordA, err
 }
+func (objMgr *ObjectManager) GetARecord(recordName string,  ipAddr string) ([]RecordA,error){
```

Comment:
If this is only getting A records by `ipaddr` , rename the method that way. `GetARecord` in general should have only `mandatory` fields

Reply:
better to use `opts`
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis